contrib/pcap2emulation.py

```diff
@@ -0,0 +1,361 @@
+                                        continue
+                                    event_str = {
+                                        "Id": "GetStringDescriptor:DescIndex=0x{:02x}".format(
+                                            desc_index
+                                        )
+                                    }
+                                    # duplicate the event so it can also be used for GetStringDescriptorBytes
+                                    language_id = get_int(layers["usb_usb_LanguageId"])
+                                    length = get_int(layers["usb_usb_setup_wLength"])
+                                    event_bytes = {
+                                        "Id": "GetStringDescriptorBytes:DescIndex=0x{:02x}".format(
+                                            desc_index
+                                        )
+                                    }
+                                    event_bytes["Id"] += ",Langid=0x{:04x}".format(
+                                        language_id
+                                    )
+                                    event_bytes["Id"] += ",Length=0x{:x}".format(length)
+
+                                elif "usb_usb_bString" in layers:
+                                    # Found a new STRING DESCRIPTOR response
+                                    data = (
+                                        layers["usb_usb_bString"].encode("utf-8").hex()
+                                    )
+                                    if "usb_usb_capdata" in layers:
+                                        # Add leftover capture data
+                                        data += "00"
+                                        data += layers["usb_usb_capdata"].replace(
+                                            ":", ""
+                                        )
+                                    event_str["Data"] = str(
+                                        base64.b64encode(bytes.fromhex(data)), "utf-8"
+                                    )
+
+                                    # now that the event is completed it can be added to the device events
+                                    self._save_event(event_str)
+                                    # duplicate the event so it can also be used for GetStringDescriptorBytes
+                                    event_bytes["Data"] = event_str["Data"]
+                                    self._save_event(event_bytes)
+
+                            if descriptor_type == DESCRIPTOR_INTERFACE:
+                                interface = {
+                                    "Length": get_int(layers["usb_usb_bLength"][-1]),
+                                    "DescriptorType": 4,
+                                }
+                                val = get_int(layers["usb_usb_bInterfaceNumber"])
+                                if val != 0:
+                                    interface["InterfaceNumber"] = val
```

Comment:
stuff like this seems to beg a dictionary matching table.    Something like this:
```
table = {
"usb_usb_bInterfaceNumber" : "InterfaceNumber"
}
for o in table:
    val = get_int(layers[o])
    if val != 0:
        interface[o[table]] = val
```